worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,386 @@
+            self.multiworld.get_location("Flip Cards", self.player).place_locked_item(self.create_item("Bonus Consumables"))
+            self.multiworld.get_location("Drawing Lots", self.player).place_locked_item(self.create_item("Bonus Consumables"))
+            self.multiworld.get_location("Match Cards", self.player).place_locked_item(self.create_item("Bonus Consumables"))
+
+
+        excluded_items = self.get_excluded_items()
+
+        pool = self.get_item_pool(self.player, excluded_items)
+
+        self.generate_filler(self.multiworld, self.player, pool)
+
+        self.multiworld.itempool += pool
+
+    def generate_output(self, output_directory: str) -> None:
+
+        rompath = ""  # if variable is not declared finally clause may fail
+        try:
+            world = self.multiworld
+            player = self.player
+            rom = LocalRom(get_base_rom_path())
+            patch_rom(self, rom, self.player)
+
+            rompath = os.path.join(output_directory,
+                                   f"{self.multiworld.get_out_file_name_base(self.player)}.sfc")
+            rom.write_to_file(rompath)
+            self.rom_name = rom.name
+
+            patch = YIDeltaPatch(os.path.splitext(rompath)[0]+YIDeltaPatch.patch_file_ending,
+                                  player=player, player_name=world.player_name[player],
+                                  patched_path=rompath)
+            patch.write()
+        finally:
+            self.rom_name_available_event.set()
+            if os.path.exists(rompath):
+                os.unlink(rompath)
+
+    def modify_multidata(self, multidata: dict) -> None:
+        import base64
+        # wait for self.rom_name to be available.
+        self.rom_name_available_event.wait()
+        rom_name = getattr(self, "rom_name", None)
+        if rom_name:
+            new_name = base64.b64encode(bytes(self.rom_name)).decode()
+            multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]
+
+    def extend_hint_information(self, hint_data: typing.Dict[int, typing.Dict[int, str]]) -> None:
+        stage_pos_data = {}
+        for loc in self.multiworld.get_locations(self.player):
+            if loc.LevelID is not None and loc.address is not None:
+                if loc.LevelID in self.world_1_stages:
+                    stage_pos_data[loc.address] = "World 1"
+                elif loc.LevelID in self.world_2_stages:
+                    stage_pos_data[loc.address] = "World 2"
+                if loc.LevelID in self.world_3_stages:
+                    stage_pos_data[loc.address] = "World 3"
+                if loc.LevelID in self.world_4_stages:
+                    stage_pos_data[loc.address] = "World 4"
+                if loc.LevelID in self.world_5_stages:
+                    stage_pos_data[loc.address] = "World 5"
+                if loc.LevelID in self.world_6_stages:
+                    stage_pos_data[loc.address] = "World 6"
```

Comment:
```suggestion
            level_id = getattr(loc, "LevelID")
            for level, stages in zip(["World {i}" for i in range(1, 7)], [self.world_1_stages, self.world_2_stages, 
                                                                          self.world_3_stages, self.world_4_stages, 
                                                                          self.world_5_stages, self.world_6_stages]):
                if level_id in stages:
                    stage_pos_data[loc.address] = level
                    break
```
The small code boxes Github has makes the spacing look bad.